test/system/runScenario3.js

```diff
@@ -0,0 +1,26 @@
+const runScenario3 = (trader2) => {
+  console.log('********************* Scenario 3 :  Test balance checks *********************');
+  return trader2.placeLimitOrder('S', trader2.getBalance().assets + 25, 0.14)
+    .then(() => { throw new Error('The function resolved when it shouldn\'t have');},
+      (err) => console.log(err.response.data.message))
+    .then(() => trader2.placeLimitOrder('S', ((trader2.getBalance().assets/2)), 0.14))
+    .then(() => trader2.placeLimitOrder('S', ((trader2.getBalance().assets * 2) * 3 / 4), 0.14))
+    .then(() => { throw new Error('The function resolved when it shouldn\'t have');},
+      (err) => { console.log(err.response.data.message);})
```

Comment:
There is a subtle bug in the the scenario3 code that might confuse things:
If the first attempt to place an order on line 3 succeeds, then the resolve handler on line 4 will throw an exception('the function resolved...').
This exception will be propagated down the Promise chain until it hits this .then on lines 8-9 where it will be handled by the reject handler (line 9) where it will be console logged and then ignored.

There are a few ways to mitigate this, but the most useful one (for this system test) is add exception expectations to the Trader functions, so lines 3-9 can be replaced with 
```javascript
  return trader2.placeLimitOrder('S', trader2.getBalance().assets + 25, 0.14, { expectFail: true })
    .then(() => trader2.placeLimitOrder('S', ((trader2.getBalance().assets/2)), 0.14)) // default behaviour is expectFail: false
    .then(() => trader2.placeLimitOrder('S', ((trader2.getBalance().assets * 2) * 3 / 4), 0.14, { expectFail: true ))
    // ...
```

The Trader class function createOrder can change to 
```javascript
createOrder(price side, quantity, { expectFail = false }) {
  const market = this.market;
  const type = 'L'; // this should really be a function argument
  const order = {market, side, type, quantity, price };
  // note - since client.createOrder returns a Promise, we don't need to wrap it inside another one
  return this.client.createOrder(order)
    .then(
      value => {
        if (expectFail) {
          throw new Error('createOrder succeeded when expected to fail');
        }
        return value;
      },
      error => {
        if (expectFail) {
          // ignore this error, it was expected.
          return;
        }
        throw error;
      }
    );
}
```

The placeLimitOrder can simply pass the options argument directly to createOrder and the requests should work as expected